### PR TITLE
CRM-20750:Incorrect financial trxn entries when payment instrument is changed on backoffice Contribution edit form

### PR DESCRIPTION
**Pull request by monishdeb**
* [CRM-20750: Incorrect financial trxn entries when payment instrument is changed on backoffice Contribution edit form ](https://issues.civicrm.org/jira/browse/CRM-20750)

### CODE REVIEW

**Comment by monishdeb** [comment]
Test this please

**Comment by monishdeb** [comment]
Test this please

File: CRM/Financial/BAO/FinancialItem.php [hunk 1]
```diff
@@ -149,6 +149,8 @@ public static function add($lineItem, $contribution, $taxTrxnID = FALSE, $trxnId
    * @return CRM_Financial_DAO_FinancialItem
    */
   public static function create(&$params, $ids = NULL, $trxnIds = NULL) {
+    CRM_Core_Error::backtrace('dsfds', TRUE);
+    CRM_Core_Error::debug_var('$params', $params);
```
Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialItem.php]: Backtrace????

**Comment by monishdeb** [comment]
@eileenmcnaughton this fixes the bug mentioned in the JIRA and also involves moving the ```changePaymentInstrument``` code in a proper fn. Can you please check my PR?

**Comment by JoeMurray** [comment]
QA'd and looks good now:

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by eileenmcnaughton** [comment]
@monishdeb should we make this WIP pending review of the other PR (which also fixes this from my testing )

**Comment by monishdeb** [comment]
@eileenmcnaughton as per #10776 it doesn't allow us to change PI via the edit form. But what if PI is changes via contribution.update API. So in order to fix the underlying code to avoid such oddities in financial trxn I think this fix is required. What do you say?

**Comment by eileenmcnaughton** [comment]
OK - I can see there are api test changes too - so I see what you are saying - will find more time to look more carefully.

**Comment by monishdeb** [comment]
Closing in favor of #10920

**Pull request closed without merging**